ambry-tools/src/main/java/com/github/ambry/clustermap/HelixVcrPopulateTool.java

```diff
@@ -92,11 +101,26 @@ public static void main(String[] args) {
     }
 
     if (options.has(updateClusterOpt)) {
-      String srcZkString = options.valueOf(srcOpt).split(SEPARATOR)[0];
-      String srcClusterName = options.valueOf(srcOpt).split(SEPARATOR)[1];
-      System.out.println("Updating cluster: " + destClusterName + "by checking " + srcClusterName);
       boolean dryRun = options.has(dryRunOpt);
-      updateResourceAndPartition(srcZkString, srcClusterName, destZkString, destClusterName, dryRun);
+      if (options.has(srcOpt)) {
+        String[] srcZkAndCluster = options.valueOf(srcOpt).split(SEPARATOR);
+        if (srcZkAndCluster.length != 2) {
+          errorAndExit("src argument must have form 'zkString/clusterName'");
+        }
+        String srcZkString = srcZkAndCluster[0];
+        String srcClusterName = srcZkAndCluster[1];
+        System.out.println("Updating cluster: " + destClusterName + "by checking " + srcClusterName);
+        updateResourceAndPartition(srcZkString, srcClusterName, destZkString, destClusterName, dryRun);
+      } else {
+        // Just update cluster config, not resources
```

Comment:
I am a little confused about this comment because I see resource ideal state is updated in the following lines.

Reply:
Yeah, I meant the resources would not be updated with any new partitions, though their idealstate is still updated.  I removed the comment, it's not needed.